### PR TITLE
Next.js Migration

### PR DESCRIPTION
**Pull request by zachdt**
Includes state changes in #23 

This PR includes architecture for using next.js SSR alongside Wagmi.  Most of the huge migration issues are resolved.

Pages and their relevant views are currently affected by the lack of state/context functionality.  Tons of TODO here, including adding loading state to each context and default values.  Context update hooks need to be implemented in pages/components before pages will work properly.

Let's begin the refactor starting with the base route. Old pages are still there under `src/pages/legacy`.  We do not require a list of all pools at the moment, lets focus purely on styling for now.  I will continue to build out the state management before working on product details.

We can discuss specifications for this Friday when it is more detailed and convenient.  For now, lets just mock the "issuance" and "redeemption" pages without state.  Base route page should just be two buttons selecting between these two routes.

Please contact me with any questions.

### CODE REVIEW

**Comment by zachdt** [comment]
Vercel has been updated to use Next.js and new environment variables.  New deployments made w/ Vite will no longer work

**Comment by zachdt** [comment]
repairing vercel build, cleaning up the new pages

**Comment by zachdt** [comment]
legacy pages are filled with typeerrors.  wont refactor anything else there until i meet with clement to discuss next steps.  i can manually force a next.js build, but moving forward we should comply with enchanced typesafety requirements.

**Comment by zachdt** [comment]
fixed tailwind!

**Comment by zachdt** [comment]
Next.js requires Suspense error handling for async, app currently breaks instantly.  Adding error handling now... very close!

**Comment by zachdt** [comment]
Added Page-level error tracking with Suspense, app is now stable in development! 

Still need to fix legacy page type-errors to get production deployments cooking before merge, but this is a good spot to wrap up for now.

@clemlak let me know when u are available either in the next few hours or in the early morning.  We should be able to easily grind out the type-errors and get us setup to work quickly with Matt over the next week.

File: src/pages/legacy/home/index.tsx [hunk 1]
```diff
@@ -71,7 +80,9 @@ function Home() {
                       </div>
                     </div>
                   </td>
-                  <td className="text-right">${(pool.reserveX * prices[pool.tokenX.symbol] + pool.reserveY * prices[pool.tokenY.symbol]).toLocaleString(undefined)}</td>
+                  <td className="text-right">
+                  ${(pool.reserveX * prices[pool.tokenX.symbol] + pool.reserveY * prices[pool.tokenY.symbol]).toLocaleString(undefined)}
```
Comment (Alexangelj) [inline src/pages/legacy/home/index.tsx:84]: put this in a variable above the return?

**Review by Alexangelj — COMMENTED** [review]
So just need to get around the exponentiation thing with the bigint to get this to build?

File: src/pages/legacy/home/index.tsx (hunk 1 above)
Comment (zachdt) [reply src/pages/legacy/home/index.tsx:84]: Its a bit more complex, the state context isn't strictly typed and we have to build out error boundaries.  the app wont build due to page type-errors, but it also crashes outside those views due to lack of error mgmt.   matt and I are going to be working thru some refactors to reflect the base-app, this pr will change a lot over the next day.

Comment (zachdt) [reply src/pages/legacy/home/index.tsx:84]: previously, react would just steamroll thru those crashes but we need stability

**Comment by kinrezC** [comment]
we're no longer pursuing the next migration, closing.

**Pull request closed without merging**